### ISSUE TITLE
repeat operator not applied consistently

### ISSUE THREAD

**Issue by seanyoung** [bengtmartensson/IrpTransmogrifier#239]
Consider the following:
```bash
$ ./irptransmogrifier.sh --irp '{100}<1,-1|1,-3>((16,-5m)+)' render -r 
Freq=38000Hz[+1600,-5000][+1600,-5000][]
```
Now if we add a (unused) bitspec to the `(15,-5m)+`, suddenly the repeat changes:
```bash
 ./irptransmogrifier.sh --irp '{100}<1,-1|1,-3>(<1|3>(16m,-5m)+)' render -r 
Freq=38000Hz[+16000,-5000][][]
```

**Comment by bengtmartensson** [comment]
Sean, although this did not make it to the just released 1.2.12, I will address it shortly.

**Comment by seanyoung** [comment]
> Turn out this was much more of a problem than I thought. I have rewritten much of the extract functionality (`IrStream.extractPass(IrSignal.Pass)`). Checked in to the branch refactor-extraction.
> 
> Sean, I want to ask you to kindly review the commits. See in particular the tests in IrStreamNGTest.

Thanks. I've gone through the code once, and run some tests. Looks good so far.

> I have not said it anywhere else, but variations in IrSequences having a finite upper bound will not work as in Graham's specification -- I should document this.

What's the difference?
